webui/react/src/hooks/useModal/UserSettings/useModalChangeName.test.tsx

```diff
@@ -0,0 +1,131 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Button, Modal } from 'antd';
+import React, { useCallback } from 'react';
+
+import StoreProvider, { StoreAction, useStoreDispatch } from 'contexts/Store';
+import { PatchUserParams } from 'services/types';
+import { DetailedUser } from 'types';
+
+import useModalUserSettings from './useModalUserSettings';
+
+jest.mock('services/api', () => ({
+  patchUser: ({ userParams }: PatchUserParams) => {
+    return Promise.resolve({ displayName: userParams.displayName });
```

Comment:
nit: this should return the whole user object, not just the display name. I think it also needs to update the value in the store.